LocalPackages/DataBrokerProtection/Sources/DataBrokerProtection/Operations/DataBrokerOperation.swift

```diff
@@ -215,3 +215,40 @@ class DataBrokerOperation: Operation, @unchecked Sendable {
     }
 }
 // swiftlint:enable explicit_non_final_class
+
+extension Array where Element == BrokerJobData {
+    /// Filters jobs based on their preferred run date:
+    /// - Opt-out jobs with no preferred run date are included.
+    /// - Jobs with a preferred run date on or before the priority date are included.
+    ///
+    /// Note: Opt-out jobs without a preferred run date may be:
+    /// 1. From child brokers (will be skipped during runOptOutOperation).
+    /// 2. From former child brokers now acting as parent brokers (will be processed if extractedProfile hasn't been removed).
+    func filtered(using priorityDate: Date) -> [BrokerJobData] {
```

Comment:
I think it would be good to rename this so it better conveys what it does.
Am I right in thinking it removes any with a perferredrundate after the priorityDate? (and keeps all the rest)

Reply:
Correct. preferredRunDate after priorityDate are removed. 
Renamed to `eligibleForRun(byDate:)`